components/ws-proxy/pkg/sshproxy/server.go

```diff
@@ -318,29 +318,30 @@ func (s *Server) HandleConn(c net.Conn) {
 		OwnerUserId: wsInfo.OwnerUserId,
 	}
 
-	if wsInfo.SSHKey != nil {
-		key, err = ssh.ParsePrivateKey([]byte(wsInfo.SSHKey.Private))
-		if err != nil {
+	if !wsInfo.IsManagedByMk2 {
+		if s.caKey == nil || !wsInfo.IsEnabledSSHCA {
+			err = xerrors.Errorf("workspace not managed by mk2, but didn't have SSH CA enabled")
+			s.TrackSSHConnection(wsInfo, "connect", ErrCreateSSHKey)
+			ReportSSHAttemptMetrics(ErrCreateSSHKey)
+			log.WithField("instanceId", wsInfo.InstanceID).WithError(err).Error("failed to generate ssh cert")
```

Comment:
Question, non-blocking:
Should we also include `wsInfo.IsManagedByMk2` as a field? I ask because I'm unsure if a nextgen cluster will support and have both managers at the same time. I assume it will.